tests/test_compiler.py

```diff
@@ -0,0 +1,346 @@
+# Copyright 2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for the compiler subpackage.
+TODO: Uncomment 'pytest.mark.external' to check these tests in GitHub actions with
+    the 'pennylane-catalyst' v0.3.2 release. These tests require the installation
+    of Catalyst from the main branch at the moment.
+"""
+# pylint: disable=import-outside-toplevel
+import pytest
+import pennylane as qml
+
+from pennylane import numpy as np
+
+catalyst = pytest.importorskip("catalyst")
+jax = pytest.importorskip("jax")
+
+# pytestmark = pytest.mark.external
+
+from jax import numpy as jnp  # pylint:disable=wrong-import-order, wrong-import-position
+from jax.core import ShapedArray  # pylint:disable=wrong-import-order, wrong-import-position
+
+# pylint: disable=too-few-public-methods, too-many-public-methods
+
+
+class TestCatalyst:
+    """Test ``qml.qjit`` with Catalyst"""
+
+    def test_compiler(self):
+        """Test compiler active and available methods"""
+
+        assert not qml.compiler.active()
+
+        assert qml.compiler.available("catalyst")
+        assert qml.compiler.available_compilers() == ["catalyst"]
+
+        assert qml.compiler.available("catalyst")
+        assert qml.compiler.available_compilers() == ["catalyst"]
```

Comment:
unless you're testing that it's stable?
```suggestion
```

Reply:
yeah, it's part of my first proposed implementation using global variables for keeping record of installed compiler packages. It's not required anymore. thanks!